packages/aws-cdk/lib/api/aws-auth/sdk.ts

```diff
@@ -6,21 +7,10 @@ import { cached } from './cached';
 import { Account } from './sdk-provider';
 import { traceMethods } from '../../util/tracing';
 
-// We need to map regions to domain suffixes, and the SDK already has a function to do this.
-// It's not part of the public API, but it's also unlikely to go away.
-//
-// Reuse that function, and add a safety check, so we don't accidentally break if they ever
-// refactor that away.
-
 /* eslint-disable @typescript-eslint/no-require-imports */
-const regionUtil = require('aws-sdk/lib/region_config');
 require('aws-sdk/lib/maintenance_mode_message').suppress = true;
 /* eslint-enable @typescript-eslint/no-require-imports */
 
-if (!regionUtil.getEndpointSuffix) {
```

Comment:
Our unit tests mock this function so it can be tested here. Please remove this mock from those unit tests.

Reply:
Originally I removed this but it caused `test/api/logs/logs-monitor.test.ts` to fail.

I have it passing by changing the expected num of stderr errors to 2, and pointing the check of the error content to the second index. Let me know if there's a different approach you would rather I take with this.

Reply:
So you removed the mock and that caused that test to fail? What did it fail with?

Reply:
When removing the line `require('aws-sdk/lib/maintenance_mode_message').suppress = true;`, the test suite fails with a message that it was expecting one stderr message. I have left the test in that was failing, and have got it to pass by setting it to expect 2 stderr messages

This commit passes the tests without any tests / mocks being removed:
https://github.com/aws/aws-cdk/pull/31008/commits/07d4e30e3b316d9d28eef97b6cb6fdd55abd35aa#